src/Models/Services/MsalAuthService.ts

```diff
@@ -149,58 +268,130 @@ export default class MsalAuthService implements IAuthService {
+    };
+
+    private getAdtToken = (): Promise<string> => {
+        const adtApiEndpoint = this.apiEndpoints.adt;
+        return new Promise(
+            this.getGenericTokenPromiseCallback({
+                scopes: [`${adtApiEndpoint}/.default`]
+            })
+        );
+    };
+
+    private getAdxToken = (): Promise<string> => {
+        const adxApiEndpoint = this.apiEndpoints.adx;
+        return new Promise(
+            this.getGenericTokenPromiseCallback({
+                scopes: [`${adxApiEndpoint}/user_impersonation`]
+            })
+        );
+    };
+
+    private getStorageToken = (): Promise<string> => {
+        const storageApiEndpoint = this.apiEndpoints.storage;
+        return new Promise(
+            this.getGenericTokenPromiseCallback({
+                scopes: [`${storageApiEndpoint}/user_impersonation`]
+            })
+        );
+    };
+
+    private getManagementToken = (
+        useLoginAuthority = false
+    ): Promise<string> => {
+        const managementApiEndpoint = this.apiEndpoints.management;
+        const scope = { scopes: [`${managementApiEndpoint}/.default`] };
+        if (useLoginAuthority) {
+            scope['authority'] = `${this.authParams.login}organizations`;
         }
+        return new Promise(this.getGenericTokenPromiseCallback(scope));
+    };
+
+    private getPowerBIToken = (): Promise<string> => {
+        const powerBIApiEndpoint = this.apiEndpoints.powerBI;
+        const scope = { scopes: [`${powerBIApiEndpoint}/.default`] };
+        return new Promise(this.getGenericTokenPromiseCallback(scope));
+    };
+
+    private promiseHttpRequest = (token, url, payload, verb = 'POST') =>
+        new Promise((resolve, reject) => {
+            const xhr = new XMLHttpRequest();
+            xhr.onreadystatechange = () => {
+                if (xhr.readyState !== 4) {
+                    return;
+                }
+
+                if (xhr.status === 200 || xhr.status === 202) {
+                    if (xhr.responseText.length === 0) {
+                        resolve({});
+                    } else {
+                        resolve(xhr.responseText);
+                    }
+                } else {
+                    reject(xhr);
+                }
+            };
+            xhr.open(verb, url);
+            xhr.setRequestHeader('Authorization', `Bearer ${token}`);
+            xhr.send(payload);
+        });
+
+    ssoSilent = (userName, continuation?: () => void) => {
+        const request: SsoSilentRequest = {
+            authority: `${this.authParams.login}${this.tenantId}`,
+            loginHint: userName
+        };
+        const setAccountAndContinue = (response) => {
+            if (response.account) {
+                this.setActiveAccount(response.account);
+            }
+            continuation?.();
+        };
+        const catchLogic = (err) => {
+            console.error('SSO failed');
+            console.error(err);
+            continuation?.();
```

Comment:
why do we call continuation when we fail?

Reply:
this catch logic is called if the ssosilent fails with error which is not type of "InteractionRequiredAuthError" which triggers popup login. In all other cases, if ssosilent fails with other type of errors, we should not break the flow if user is setting "isloading" to false etc. in the continuation as we do in the public repo fork. @darsney can confirm though.